napari/components/dims.py

```diff
@@ -225,6 +231,53 @@ def set_range(
                     ax = assert_axis_in_bounds(int(ax), self.ndim)
                     full_range[ax] = r
                 self.range = full_range
+        if restore_point is not None:
+            new_point = self._get_point_in_ndim(restore_point)
+            self.set_point(range(self.ndim), new_point)
+
+    @staticmethod
+    def rounded_division(min_val, max_val, precision):
+        return int(((min_val + max_val) / 2) / precision) * precision
+
+    def _get_point_in_ndim(
+        self,
+        point: Union[Union[int, float], Sequence[Union[int, float]]],
```

Comment:
I don't think the code handles scalars as it calls `len`. As the method is private, just simplify the. type

```suggestion
        point: Sequence[Union[int, float]],
```

Reply:
sequence only needs to support `__getitem__` with int key, there is no requirement to accept slice. 

https://docs.python.org/3/glossary.html#term-sequence

If we specify to accept float then it also will accept int. So it could be changed to `Sequence[float]`

Reply:
I changed it to be `Sequence[float]`